components/server/src/ome/services/ThumbnailBean.java

```diff
@@ -819,10 +817,18 @@ public void createThumbnail(Integer sizeX, Integer sizeY)
         }
     }
 
-    /** Actually does the work specified by {@link #createThumbnail(Integer, Integer)}. */
+    /** Helper function for using local thumbnailMetadata */
     private Thumbnail _createThumbnail() {
+        // For old times sake
+        return _createThumbnail(thumbnailMetadata);
+    }
+
+    /*
+     * Actually does the work specified by {@link #createThumbnail(Integer, Integer)}.
+     */
+    private Thumbnail _createThumbnail(Thumbnail thumbMetaData) {
         StopWatch s1 = new Slf4JStopWatch("omero._createThumbnail");
-        if (thumbnailMetadata == null) {
+        if (thumbMetaData == null) {
```

Comment:
Why did this variable get changed?

Reply:
It's now a local variable for the method. A new method that passes in thumbnailMetadata as a parameter was added.